shopify_function_macro/src/lib.rs

```diff
@@ -272,6 +272,18 @@ pub fn typegen(
     module.to_token_stream().into()
 }
 
+/// Helper function to determine if a GraphQL input field type is nullable
+/// Uses conservative detection to identify Optional fields from GraphQL schema
+fn is_input_field_nullable(ivd: &impl InputValueDefinition) -> bool {
+    // Use std::any::type_name to get type information as a string
+    let type_name = std::any::type_name_of_val(&ivd.r#type());
```

Comment:
Not super familiar with bluejay, but I think if we have something like:

```
ivd.r#type().is_nullable() {
  ```
  
Then we could avoid using this approach. We need to get the type information so we can match correctly. I ran into issues when trying to use matching because some types were required (aka not optional), and some were optional.

Reply:
Yeah we should be using `bluejay` here for the type information